Modules/_datetimemodule.c

```diff
@@ -3259,19 +3259,30 @@ date_fromtimestamp(PyObject *cls, PyObject *obj)
 static PyObject *
 date_today(PyObject *cls, PyObject *Py_UNUSED(dummy))
 {
-    PyObject *time;
-    PyObject *result;
-    time = time_time();
-    if (time == NULL)
+    /* Use C implementation to boost performance for date type */
+    if ((PyTypeObject *)cls == &PyDateTime_DateType) {
+        struct tm tm;
+        time_t t;
+        time(&t);
+
+        if (_PyTime_localtime(t, &tm) != 0) {
+            return NULL;
+        }
+
+        return new_date_ex(tm.tm_year + 1900,
+                           tm.tm_mon + 1,
+                           tm.tm_mday,
+                           (PyTypeObject*)cls);
+    }
+    PyObject *time = time_time();
```

Comment:
```suggestion
    }

    PyObject *time = time_time();
```

For an improved clarity